specification.markdown

```diff
@@ -155,11 +155,39 @@ Class D    | 4      | 0x101FFFFF - 0x1FFFFFFE |   268,338,304
+## Unknown Data Size
+
+An Element Data Size with all VINT_DATA bits set to one is reserved as an indicator that the size of the EBML Element is unknown. The only reserved value for the VINT_DATA of Element Data Size is all bits set to one. An EBML Element with an unknown Element Data Size is referred to as an Unknown-Sized Element. A Master Element MAY be an Unknown-Sized Element; however an EBML Element that is not a Master Element MUST NOT be an Unknown-Sized Element. Master Elements MUST NOT use an unknown size unless the unknownsizeallowed attribute of their EBML Schema is set to true (see [the section on the unknownsizeallowed attribute](#unknownsizeallowed)).
+
+The use of Unknown-Sized Elements allows for an EBML Element to be written and read before the size of the EBML Element is known. Unknown-Sized Element MUST NOT be used or defined unnecessarily; however if the Element Data Size is not known before the Element Data is written, such as in some cases of data streaming, then Unknown-Sized Elements MAY be used. The end of an Unknown-Sized Element is determined by whichever comes first:
+
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a Parent Element of the Unknown-Sized Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is not a Descendant Element of the Unknown-Sized Element but share a common direct parent. For example a Top-Level Element.
+- Any valid EBML Element according to the EBML Schema, Global Elements excluded, that is a root element.
+- The end of the EBML Document, either when reaching the end of the file or because a new EBML Header started.
+
+Consider an Unknown-Size Element which EBML path is `\root\level1\level2\elt`. Here are some possible and impossible way to end this element:
+
+EBML Path of next element          | Status
+-----------------------------------|------------------------------
+`\root\level1\level2`              | Ends the unknown-size element, as it is a new Parent element
+`\root\level1`                     | Ends the unknown-size element, as it is a new Parent element
+`\root`                            | Ends the unknown-size element, as it is a new root element
+`\root2`                           | Ends the unknown-size element, as it is a new root element
+`\root\level1\level2\other`        | Ends the unknown-size element, as they share the same parent
+`\root\level1\level2\elt\inside`   | Doesn't end the unknown-size element, it's a child of `elt`
+`\root\level1\level2\other\part`   | Isn't valid right after `elt`. `other` should be found first
```

Comment:
From the description, it's unclear if you imply that `part` signifies the end of `elt`. I think not, but it's unclear. For instance if you have `part` as the first part of the data of `elt`, then is `elt` ended as considered empty, then `part` is considered misplaced?

Reply:
It's more a reminder that you can't skip levels between elements. You have to have the proper parent first. 

It may be a bit confusing because the decision to end an Unknown-Sized Element happens when the ID of an new element is read, not when the whole next element is read.

In this case you can't get the ID for `part` if you don't have the ID for `other` first. This case is covered 2 lines above.

Reply:
I'll remove this line. It doesn't help understand anything not covered above.